### ISSUE TITLE
Improve screen reader and accessibility support

### ISSUE THREAD

**Issue by francipvb** [sqlitebrowser/sqlitebrowser#1451]
Hello,

Screen reader support needs to be improved. In various points it is impossible to perform basic operations such add a table. I mean the table can be added, but the column properties cannot be changed easily.

In fact, accessibility is like visual design, but in terms on control exposed accessible information.

The first place to start is in QT framework version used to build the project.

In QT 5.11 the bridge for accessibility starts to implement a more native accessibility interface, that is, UIAutomation plus a bunch of improvements in the QT's core accessibility API.

You can test with [NVDA](https://www.nvaccess.org/).

i think that this can be used as a master issue for various tasks related to base accessibility support.

I think that I'll perform some tests under fedora these days for issues related directly with ATSPI implementation problems.

Cheers,

**Comment by justinclift** [comment]
Good thinking @francipvb. :smile:

It's interesting that Qt 5.11 might start to make this practical.  For our windows builds, we use Qt 5.7 because that's the last version that will compile code to be Win XP compatible.  And yeah, there does seem to be a small (but persistent) group of people still using that OS. :wink:

That's only really needed for our 32-bit builds on Windows though.  For the 64-bit builds, we've been using Qt 5.7 as well just to match, but it shouldn't be much of a problem for us to upgrade to something newer.

eg keep using Qt 5.7 for the 32-bit builds, but potentially use Qt 5.11 for the 64-bit ones.

Guessing that would at least help get things on the right track. :smile:

Our macOS builds are currently using Qt 5.9.2.  Not for any special reason (that I remember), more just that was the latest version at the time when I updated Qt on the system.  Shouldn't be hard to update that to Qt 5.11 either. :smile:

**Comment by justinclift** [comment]
No worries.  I should be able to update the Qt version for our 64-bit Windows compiles today.

@francipvb If I get that done, would you have the time and interest to try out our automatic nightly build (Windows 64-bit) tomorrow, and let us know if that improves things?

**Comment by francipvb** [comment]
Hello,

I can, but at night. I'll have a connection to do some testing and respond to you.
Cheers,



Enviado desde mi Samsung Mobile de Claro


-------- Mensaje original --------
De: Justin Clift <notifications@github.com>
Fecha: 12/7/18 3:39 PM (GMT-03:00)
A: sqlitebrowser/sqlitebrowser <sqlitebrowser@noreply.github.com>
CC: "Francisco R. Del Roio" <francipvb@hotmail.com>, Mention <mention@noreply.github.com>
Asunto: Re: [sqlitebrowser/sqlitebrowser] Improve screen reader and accessibility support (#1451)


No worries. I should be able to update the Qt version for our 64-bit Windows compiles today.

@francipvb<https://nam04.safelinks.protection.outlook.com/?url=https%3A%2F%2Fgithub.com%2Ffrancipvb&data=02%7C01%7C%7C9d681c7b836c478520c808d5e826ca09%7C84df9e7fe9f640afb435aaaaaaaaaaaa%7C1%7C0%7C636670175643298481&sdata=bTRrRcoi%2FDbpfydJy5UgrVAxEH2Eiy9hp9SUF1yYbYg%3D&reserved=0> If I get that done, would you have the time and interest to try out our automatic nightly build (Windows 64-bit) tomorrow, and let us know if that improves things?

**Comment by justinclift** [comment]
That's no problem at all.  It'll get the update done in the next few hours, and the automatic build should pull that in overnight and put it here:

&nbsp; &nbsp; https://nightlies.sqlitebrowser.org/latest/

We're also trying out a fix for a windows build problem, which is why there aren't any Windows builds in that directory at the moment.  In theory (if that fix works), they should be there tomorrow.  And using Qt 5.11.  So they'll be ready for you to test when you have the time. :smile:

**Comment by justinclift** [comment]
Well, it turns out that we can't upgrade the Qt version on the build server to 5.11.x yet.

Our build system still uses MSVC 2013, and support for that was dropped with Qt 5.11.

I've just updated the build server to Qt 5.10.1 for our 64-bit builds, just to see if that works out any better.  So, there probably isn't much for @francipvb to test tomorrow after all.

It'll have to wait until I update to a newer MSVC version, which will likely take a fair bit of effort as (I think) we will need to change our SQLCipher and TCL build steps as well.

**Comment by justinclift** [comment]
There's not really anything to test yet.  The upgrade of Qt didn't go well, so the highest version we were able to easily get to was Qt 5.8.  That's a bit higher than the Qt 5.7 we were previously using, but it's not real close to the desired Qt 5.11.

It's probably going to take a fair bit of effort to upgrade all of the pieces which Qt 5.11 needs, so it's not likely to happen in the near future. At some point it definitely will though.

**Comment by mgrojo** [comment]
By the way, while I was trying to analyse #1491 I got this message in the terminal.

```
Problem creating accessible interface for:  QObject(0x310cee0) 
Make sure to deploy Qt with accessibility plugins.
```

**Comment by justinclift** [comment]
> Make sure to deploy Qt with accessibility plugins.

Oh, that's interesting.  Is that with one of our official nightly builds?

**Comment by mgrojo** [comment]
> Oh, that's interesting. Is that with one of our official nightly builds?

No, it's a compiled version from the master branch under my Ubuntu 16.04. I guess it is lacking some package, but the only relevant that I have found is qt-at-spi, and that is already installed.

By the way, I can use the program Orca to read the screen, although I don't understand much :smile:
